src/COMPASS/runconfig.py

```diff
@@ -0,0 +1,144 @@
+
+import helpers
+
+
+def validate_group(group_cfg: dict) -> None:
+    '''check groups dict'''
+    # check input group
+    input_group = group_cfg['input_file_group']
+
+    is_reference = input_group['reference_burst']['is_reference']
+    if not is_reference:
+        helpers.check_file_path(input_group['reference_burst']['file_path'])
+
+    for i, safe_file in enumerate(input_group['safe_file_path']):
+        # ensure only one safe file for reference rdr2geo processing
+        if is_reference and i > 0:
+            err_str = 'More that one safe file provided as reference'
+            logging.error(err_str)
+            raise ValueError(err_str)
+        helpers.check_file_path(safe_file)
+
+    helpers.check_file_path(input_group['orbit_file_path'])
+
+    # check dynamical ancillayr file group
+    dem_path = group_cfg['dynamic_ancillary_file_groups']['dem_file']
+    helpers.check_file_path(dem_path)
+    helpers.check_dem(dem_path)
+
+    # check product_path_group
+    product_path_group = group_cfg['product_path_group']
+    helpers.check_write_dir(product_path_group['product_path'])
+    helpers.check_write_dir(product_path_group['scratch_path'])
+    helpers.check_file_path(product_path_group['sas_output_file'])
+
+    # check polarizations
+    if 'polarization' not in group_cfg['processing']:
+        group_cfg['processing']['polarization'] = ['HH', 'HV', 'VH', 'VV']
+
+
+@dataclass(frozen=True)
+class RunConfig:
+    '''dataclass containing CSLC runconfig'''
+    name: str
+    # for easy immutability, lazily keep dict read from yaml (for now?)
+    groups: dict
+
+    @classmethod
+    def load_from_yaml(cls, yaml_path: str, workflow_name: str) -> RunConfig:
+        '''init RunConfig from yaml file'''
+        try:
+            # load schema to validate against
+            schema = yamale.make_schema(
+                f'{helpers.WORKFLOW_SCRIPTS_DIR}/schemas/{workflow_name}.yaml',
+                parser='ruamel')
+        except:
+            err_str = f'unable to load schema for workflow {workflow_name}.'
+            logging.error(err_str)
+            raise ValueError(err_str)
+
+        # load yaml file or string from command line
+        if os.path.isfile(yaml_path):
+            try:
+                data = yamale.make_data(yaml_path, parser='ruamel')
+            except yamale.YamaleError as yamale_err:
+                err_str = f'Yamale unable to load {workflow_name} runconfig yaml {yaml_path} for validation.'
+                logging.error(err_str)
+                raise yamale.YamaleError(err_str) from yamale_err
+        else:
+            raise FileNotFoundError
+
+        # validate yaml file taken from command line
+        try:
+            yamale.validate(schema, data)
+        except yamale.YamaleError as yamale_err:
+            err_str = f'Validation fail for {workflow_name} runconfig yaml {yaml_path}.'
+            logging.error(err_str)
+            raise yamale.YamaleError(err_str) from yamale_err
```

Comment:
would this provide a meaningful error message on why validation fails (in case if it fails)?

Could you use something like [this]( 
https://github.com/23andMe/Yamale#:~:text=try%3A%0A%20%20%20%20yamale.validate(schema%2C%20data)%0A%20%20%20%20print(%27Validation%20success!%20%F0%9F%91%8D%27)%0Aexcept%20YamaleError,error%20in%20result.errors%3A%0A%20%20%20%20%20%20%20%20%20%20%20%20print(%27%5Ct%25s%27%20%25%20error)%0A%20%20%20%20exit(1)) to print out the errors?

Reply:
As-is It will dump out errors in schema validation. 
![image](https://user-images.githubusercontent.com/2991893/148314768-a671fdbc-1de2-4dbd-8a74-4e4afba8ece6.png)